pkg/api/pin.go

```diff
@@ -38,14 +38,45 @@ func (s *Service) pinRootHash(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	switch err = s.pinning.CreatePin(r.Context(), paths.Reference, true); {
-	case errors.Is(err, storage.ErrNotFound):
-		jsonhttp.NotFound(w, nil)
+	putter, err := s.storer.NewCollection(r.Context())
+	if err != nil {
+		logger.Debug("pin root hash: failed to create collection", "error", err)
+		logger.Error(nil, "pin root hash: failed to create collection")
+		jsonhttp.InternalServerError(w, "pin root hash: create collection failed")
 		return
-	case err != nil:
-		logger.Debug("pin root hash: create pin failed", "chunk_address", paths.Reference, "error", err)
-		logger.Error(nil, "pin root hash: create pin failed")
-		jsonhttp.InternalServerError(w, "pin root hash: creation of tracking pin failed")
+	}
+
+	getter := s.storer.Download(true)
+	traverser := traversal.New(getter)
+
+	err = traverser.Traverse(
+		r.Context(),
+		paths.Reference,
+		func(address swarm.Address) error {
+			chunk, err := getter.Get(r.Context(), address)
+			if err != nil {
+				return err
+			}
+			err = putter.Put(r.Context(), chunk)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		err = multierror.Append(err, putter.Cleanup())
+		logger.Debug("pin collection failed", "error", multierror.Append(err, putter.Cleanup()))
```

Comment:
Not sure why is the `putter.Cleanup()` called the second time here? Use `errors.Join` instead.